src/System.Memory/src/System/SpanHelpers.byte.cs

```diff
@@ -95,82 +105,218 @@ public static int IndexOf(ref byte searchSpace, byte value, int length)
             return -1;
         }
 
-        public static bool SequenceEqual(ref byte first, ref byte second, int length)
+        public unsafe static bool SequenceEqual(ref byte first, ref byte second, int length)
         {
             Debug.Assert(length >= 0);
+            var isMatch = true;
 
-            if (Unsafe.AreSame(ref first, ref second))
-                return true;
-
-            int index = 0;
-            while (length >= 8)
+            if (length == 0)
             {
-                if (Unsafe.Add(ref first, index) != Unsafe.Add(ref second, index))
-                    return false;
-                index++;
-
-                if (Unsafe.Add(ref first, index) != Unsafe.Add(ref second, index))
-                    return false;
-                index++;
-
-                if (Unsafe.Add(ref first, index) != Unsafe.Add(ref second, index))
-                    return false;
-                index++;
-
-                if (Unsafe.Add(ref first, index) != Unsafe.Add(ref second, index))
-                    return false;
-                index++;
-
-                if (Unsafe.Add(ref first, index) != Unsafe.Add(ref second, index))
-                    return false;
-                index++;
-
-                if (Unsafe.Add(ref first, index) != Unsafe.Add(ref second, index))
-                    return false;
-                index++;
-
-                if (Unsafe.Add(ref first, index) != Unsafe.Add(ref second, index))
-                    return false;
-                index++;
-
-                if (Unsafe.Add(ref first, index) != Unsafe.Add(ref second, index))
-                    return false;
-                index++;
-
+                goto exit;
+            }
 
-                length -= 8;
+            fixed (byte* pFirst = &first)
+            fixed (byte* pSecond = &second)
+            {
+                var a = pFirst;
+                var b = pSecond;
+
+                if (a == b)
```

Comment:
nit: we call values related to the first parameter: first, a, and v0. It would be better if the names wer all realted, e.b. first, pFirst, cFirst ("cursor to first"), and vFirst ("value from first").